### ISSUE TITLE
Webhook payload stopped returning populated relations

### ISSUE THREAD

**Issue by fabianmu** [strapi/strapi#14882]
Update by: @derrickmehaffy 

**This is/was an unintended breaking change, we are looking into how to mitigate it**

---

### Required System information

- Node.js version: v16.18.0
- NPM version: 8.19.2
- Strapi version: 4.5.0
- Database: sqlite & mysql
- Operating system: macOS 13.0 (22A380)

### Describe the bug

The webhook used to populate relations in the `entry`. However that stopped and is now displayed as `count` instead.
I do not find any documentation or update in a changelog on that.

### Steps to reproduce the behavior

1. Have Content-Type with a relation
2. Enable a webook
3. Have the webhook triggered by saving a content-type
4. See related item appearing with `count` instead of the populated data in the payload of the webhook

### Expected behavior

Contain the populated relation in the webhooks payload

### Screenshots

before:
![SCR-20221115-lpc](https://user-images.githubusercontent.com/89517/202015904-233da9e7-8d12-4d93-99ec-a802f56a529f.png)
after:
![SCR-20221115-low](https://user-images.githubusercontent.com/89517/202015936-c99632e5-5897-4627-b86f-c61090d0bffe.png)

### Additional context

This used to work, I think in 4.4.7. I think the change has been introduced in 4.5.0

I'm also fine if that can be re-enabled with a configuration somewhere, which would need to be documented.

**Comment by fabianmu** [comment]
Well, it's actually better if the payload does not automatically include all related items. It could really blow up the payload.
However a way to config this would be great.

**Comment by wedneyyuri** [comment]
It's breaking our integrations, there is a way to revert this behavior?

**Comment by tefkah** [comment]
Please consider adding some way to configure this! 🙏

**Comment by derrickmehaffy** [comment]
I can confirm the payload changed, this should be a breaking change and should not be happening.

Payload on `4.5.4`:

```json
{
  "event": "entry.update",
  "createdAt": "2022-12-23T14:40:57.900Z",
  "model": "blog-post",
  "entry": {
    "id": 1,
    "test": "testtest",
    "createdAt": "2022-12-23T14:40:31.686Z",
    "updatedAt": "2022-12-23T14:40:57.857Z",
    "publishedAt": "2022-12-23T14:40:32.327Z",
    "comments": {
      "count": 1
    }
  }
}
```

Payload on `4.4.7`:

```json
{
  "event": "entry.update",
  "createdAt": "2022-12-23T14:44:39.183Z",
  "model": "blog-post",
  "entry": {
    "id": 1,
    "test": "testtest blah",
    "createdAt": "2022-12-23T14:40:31.686Z",
    "updatedAt": "2022-12-23T14:44:39.102Z",
    "publishedAt": "2022-12-23T14:40:32.327Z",
    "comments": [
      {
        "id": 1,
        "test": "test",
        "createdAt": "2022-12-23T14:40:44.565Z",
        "updatedAt": "2022-12-23T14:40:45.802Z",
        "publishedAt": "2022-12-23T14:40:45.799Z"
      }
    ]
  }
}
````

---

I won't comment on the other questions in this issue around webhooks, I would suggest taking a look at the following feature requests (feel free to vote/comment on those)

https://feedback.strapi.io/feature-requests/p/webhook-by-content-type
https://feedback.strapi.io/customization/p/customize-webhook-body
https://feedback.strapi.io/developer-experience/p/retry-webhook-requests-if-error-happens
https://feedback.strapi.io/feature-requests/p/definedisable-webhook-payload
https://feedback.strapi.io/feature-requests/p/support-discord-webhook-natively

**Comment by mariojsnunes** [comment]
this is a major breaking change! how is it still not fixed after so many months?

**Comment by Marc-Roig** [comment]
@mariojsnunes  we are close to fixing it :) 

We can not guarantee webhooks will have relations populated. 
Example:
- If the entity is created (updated, deleted, ...) using the API, not through the Admin panel. It might not have them populated if the populate parameter is not set for that.
- Same for any plugin creating the entity.

We have had some conversations about how to make webhooks behaviour more consistent. For now, we will provide a solution to make it work as before. To fix this issue.

**Comment by Marc-Roig** [comment]
this should be fixed in 4.6.1 https://github.com/strapi/strapi/pull/15363 :) 

In case you want to disable this behaviour and gain some performance, the configuration can be disabled in the `server.js` config file:

```
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: 'http://localhost:1337/',
  webhooks: {
    //  Add this to not receive populated relations in webhooks
    populateRelations: false,
  },
});
```

**Comment by fabianmu** [comment]
Awesome, thanks! And making this configurable is even better!
